tests/test_prerequisites.py

```diff
@@ -0,0 +1,272 @@
+import unittest
+from unittest import mock
+
+from pythonforandroid.prerequisites import (
+    JDKPrerequisite,
+    HomebrewPrerequisite,
+    OpenSSLPrerequisite,
+    AutoconfPrerequisite,
+    AutomakePrerequisite,
+    LibtoolPrerequisite,
+    PkgConfigPrerequisite,
+    CmakePrerequisite,
+    get_required_prerequisites,
+)
+
+
+class PrerequisiteSetUpBaseClass(object):
+    def setUp(self):
+        self.mandatory = dict(linux=False, darwin=False)
+        self.installer_is_supported = dict(linux=False, darwin=False)
+
+    def test_is_mandatory_on_darwin(self):
+        assert self.prerequisite.mandatory["darwin"] == self.mandatory["darwin"]
+
+    def test_is_mandatory_on_linux(self):
+        assert self.prerequisite.mandatory["linux"] == self.mandatory["linux"]
+
+    def test_installer_is_supported_on_darwin(self):
+        assert (
+            self.prerequisite.installer_is_supported["darwin"]
+            == self.installer_is_supported["darwin"]
+        )
+
+    def test_installer_is_supported_on_linux(self):
+        assert (
+            self.prerequisite.installer_is_supported["linux"]
+            == self.installer_is_supported["linux"]
+        )
+
+
+class TestJDKPrerequisite(PrerequisiteSetUpBaseClass, unittest.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.mandatory = dict(linux=False, darwin=True)
+        self.installer_is_supported = dict(linux=False, darwin=True)
+        self.prerequisite = JDKPrerequisite()
+
+
+class TestBrewPrerequisite(PrerequisiteSetUpBaseClass, unittest.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.mandatory = dict(linux=False, darwin=True)
+        self.installer_is_supported = dict(linux=False, darwin=False)
+        self.prerequisite = HomebrewPrerequisite()
+
+    def test_darwin_checker_is_supported(self):
+        try:
+            self.prerequisite.darwin_checker()
+        except Exception as e:
+            if e.__str__().startswith("Unsupported prerequisite check"):
+                self.fail("Darwin checker should be supported")
```

Comment:
Another option could be to leverage pytest skipif
```suggestion
    pytest.mark.skipif(sys.platform != "darwin", reason="MacOS test only")
    def test_darwin_checker_is_supported(self):
        self.prerequisite.darwin_checker()
```

Reply:
`pytest.mark.skipif(sys.platform != "darwin", reason="MacOS test only")` skips the test when not on `darwin`.

This one, instead, checks, if the `darwin_checker` is raising an error with `Unsupported prerequisite check` in it, as the `darwin_checker`, is marked as supported on this prerequisite.

Reply:
Oh right, what I don't get is why do we bother to try/except/if in this test, it's usually a red flag.
In this case why don't we simply:
```python
assert self.prerequisite.darwin_checker()
```

Reply:
With `assert self.prerequisite.darwin_checker()`, we only check if it fails (which is probably enough ATM), not specifically if it throws a `Unsupported prerequisite check on macOS for ...` exception.

BTW, re-reading the code, considering that we're testing the darwin_checker function:
https://github.com/kivy/python-for-android/blob/eb343a59896b5277ddd6057c4db0ba6573a6a74b/tests/test_prerequisites.py#L63-L69
we can probably get rid of this test?

Reply:
Ah yeah indeed, I leave that up to you either way is fine to me